docs/performance/storage_usage.rst

```diff
@@ -0,0 +1,295 @@
+|2|1704074134000|1704074399000|1|1.08|1|"N"|249|68|1|7.2|1|0.5|2.44|0|1|14.64|2.5|0|
+
+will take:
+
+- 48MB in parquet (Parquet is very optimized for storage)
+- 342MB in CSV
+- 1.2GB in json
+- 510MB in PostgreSQL 16.1 (Debian 16.1-1.pgdg120+1)
+- 775MB in CrateDB 5.9.3 (3 nodes, default values)
+
+At first sight, it might look that CrateDB storage takes more than PostgreSQL, but we need to dive deeper to really
+understand what is going on.
+
+## Table of contents
+1. [How storage works](#Storage in CrateDB)
+2. [Reducing storage](#reducing-storage)
+3. [Disable indexing](#disable-indexing)
+4. [Disable columnar store](#disable-the-columnar-store)
+5. [Changing the compression algorithm](#changing-the-compression-algorithm)
+6. [All results](#all-results)
+7. [Data normalization](#data-normalization)
+
+## How storage in CrateDB works.
+
+CrateDB is a distributed database; nodes, shards, partitions and replicas are tightly integrated.
+
+When a table is created data is sharded and distributed among nodes, this means that the memory footprint depends on
+our replication and sharding strategy.
+
+Let's break down how the `775MB` in CrateDB and the 510MB in Postgres were obtained, for `Postgres` it was straightforward:
+
+.. code-block:: sql
+SELECT pg_size_pretty( pg_total_relation_size('taxi_january') );
+
+But for CrateDB when a table is created, sharding and replication has to be taken into account.
+
+Issuing `create table my_table (my_column TEXT)` will create `max(4, num_data_nodes * 2)` shards.
+
+For example, a typical 3-node cluster, it will create:
+
+.. code-block::
+
+`max(4, 3 * 2)`
+
+`max(4, 6)`
+
+`6` shards
+
+On top of that, the default replication is the `0-1` range, so we will have a maximum of one replica, a replica multiplies the number of shards, so we will have
+`6 primary shards` and `6 replica shards`, `12 total shards` physically distributed among the three nodes.
+
+![shards.png](shards.png)
+
+You can see how many shards you have per node and table by querying the `sys.shards` table, or use this query to have a broad overlook:
+
+.. code-block:: sql
+
+SELECT
+  sum(num_docs) document_count,
+  node ['name'] node,
+  count(*) shard_count,
+  array_agg(table_name) shard_table_names
+FROM
+  sys.shards
+GROUP BY
+  node ['name']
+
+The total storage being used, can be calculated as the `avg size of 1 shard` * `total_shards`, in other words.
+
+To check this, on the current `taxi` table that was created:
+
+.. code-block:: sql
+
+SELECT
+  sum(size / 1_000_000) / count(*) as avg_mb_per_shard,
+  sum(size) / 1_000_000 as total_mb
+FROM
+  sys.shards
+WHERE
+  table_name = 'taxi'
+
+|avg_mb_per_shard|total_mb|
+ |-|-|
+|64|775|
+
+Lets check the real disk usage, by querying `select path from sys.shards` the path where the data is stored on disk can be found.
+
+```shell
+sh-5.1# pwd
+/data/data/nodes/0/indices/LeFVb9VMT_G68tZs0vOuyA
+sh-5.1# du -sh ./* | sort -h
+8.0K	./_state
+63M	./2
+63M	./3
+63M	./4
+63M	./5
+```
```

Comment:
Hi @surister. You might be more happy to write using Markdown instead of reStructuredText?

Reply:
yeah!